proto_proxy.c

```diff
@@ -2605,6 +2648,16 @@ static void proxy_process_command(conn *c, char *command, size_t cmdlen, bool mu
         return;
     }
 
+    // Count requests handled by proxy vs local.
+    // Also batch the counts down this far so we can lock once for the active
+    // counter instead of twice.
+    struct proxy_int_stats *istats = c->thread->proxy_int_stats;
+    WSTAT_L(c->thread);
```

Comment:
You mentioned in another commit that atomics are slower than locks in your benchmarking.
But curious if this causes a regression since we're adding a new lock/unlock on every command.

Reply:
There're a lot of details, but the main difference between global atomic counters and the per-worker thread mutexes are contended shared memory vs typically uncontended differential locks. These mutexes will typically never collide (only writer is that worker thread, only other time it's locked is when it's being observed), and the memory is unique to each worker thread so there's no false-sharing or shared cacheline level contention. It's complicated :)

This also isn't a new lock for this code, it was already there.